<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Workbook</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cerulean.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
}

</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Stereo-seq workshop</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="setup.html">Setup</a>
</li>
<li>
  <a href="download.html">Download</a>
</li>
<li>
  <a href="SAW.html">Workbook</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/sagc-bioinformatics/abacbs2024_workshop">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Workbook</h1>

</div>


<div id="stereo-seq-background" class="section level1">
<h1>Stereo-seq background</h1>
<p>Stereo-seq (SpaTial Enhanced Resolution Omics-sequencing) builds upon
the chip technology already used in MGI DNA nanoball sequencing
(DNBSEQ). For background on DNBSEQ, see <a
href="https://mgi-tech.eu/technology">here</a>.</p>
<p>The chips are patterned, with fixed spots of 220 nm in diameter and
spaced at a distance of 500 or 715 nm. The standard chip size is 1cm x
1cm, but the largest size available does go up to 13cm x 13cm (fresh
frozen samples only).</p>
<p>The first step is to deposit DNB templates with random barcodes on
the chip, which is sequenced to obtain the CID (coordinate ID) of each
DNB. This is the “mask file”, which is basically just a big matrix which
barcodes correspond to which locations, and is required to run SAW.</p>
<p>Then, oligos with MIDs (molecular IDs) and polyT sequences are
ligated to each spot.</p>
<p>Next, the tissue is loaded onto the chip, followed by fixation and
permeabilization to capture polyA RNAs.</p>
<p>Finally, cDNA is generated via reverse transcription and
amplification and used as a template for library prep.</p>
<p>More information on Stereo-seq can be found in the <a
href="%22https://www.cell.com/cell/fulltext/S0092-8674(22)00399-3%22">paper</a></p>
<div id="available-versions" class="section level2">
<h2>Available versions</h2>
<ul>
<li>Current fresh-frozen (FF) kit version is v1.3</li>
<li>OMNI is the new FFPE offering, which also uses total RNA</li>
</ul>
</div>
</div>
<div id="processing-data-with-saw" class="section level1">
<h1>Processing data with SAW</h1>
<p>Once your experiment is complete, you will have the following:</p>
<ul>
<li>Mask file (.h5)</li>
<li>Reads (.fastq)</li>
</ul>
<p>You will need to process the data using Stereo-seq Analysis Workflow
(SAW). This command-line tool provides similar functionality for
Stereo-seq to what Spaceranger provides for 10x spatial platforms.</p>
<p>The latest version of SAW is 8.1, with reference documentation found
here:</p>
<p><a
href="https://stereotoolss-organization.gitbook.io/saw-user-manual-v8.1"
class="uri">https://stereotoolss-organization.gitbook.io/saw-user-manual-v8.1</a></p>
<p>Actually building a reference and running SAW will take quite a long
time, so we won’t actually run it for this workshop. However, we’ll walk
through the steps and then take a look at some pre-processed files.</p>
<div id="obtaining-saw" class="section level2">
<h2>Obtaining SAW</h2>
<p>You can download the latest version of SAW from the <a
href="https://en.stomics.tech/products/stomics-software/stomics-offline-software/list.html">STOmics
download page</a> (does require making an account).</p>
<p>Alternatively, you can find earlier versions of SAW on <a
href="https://github.com/STOmics/SAW">Github</a>. Please note that the
interface has changed significantly between versions 7.x and 8.x.
Version 8.x introduces both internal workflow management with Snakemake,
and compatibility with OMNI.</p>
<p>The documentation for using SAW v8.1 with fresh-frozen datasets can
be found here: <a
href="https://stereotoolss-organization.gitbook.io/saw-user-manual-v8.1/tutorials/run-main-pipeline/stereo-seq-ff"
class="uri">https://stereotoolss-organization.gitbook.io/saw-user-manual-v8.1/tutorials/run-main-pipeline/stereo-seq-ff</a></p>
<p>In addition, <code>saw --help</code> and subcommands
(e.g. <code>saw count --help</code>) provide more information.</p>
</div>
<div id="saw-makeref" class="section level2">
<h2>SAW makeRef</h2>
<p>SAW v8.1 is quite easy to use, but not yet very configurable. To
build the reference for this example, here’s the command I used:</p>
<pre class="bash"><code>#!/bin/bash

saw makeRef \
    --mode=STAR \
    --fasta=/path/to/GRCm39.genome.fa \
    --gtf=/path/to/annotation.gtf \
    --genome=/path/to/reference/output</code></pre>
<p>Notes:</p>
<ul>
<li>The <a href="https://www.gencodegenes.org/mouse/">GENCODE GRCm39
reference</a> was used</li>
<li>The <code>--genome</code> parameter actually specifies the name of
the output directory</li>
<li>There is no way to pass parameters to STAR in this version.</li>
</ul>
<p>To have more control over the parameters used, you can use version
7.x of SAW <a href="https://github.com/STOmics/SAW">from Github</a>.
This version is implemented as a bash script which calls into a
container image, and is easier to customise (and easier to break).</p>
</div>
<div id="saw-count" class="section level2">
<h2>SAW count</h2>
<p>Next, we want to run the main pipeline which is called via
<code>saw count</code>. Here’s how it was run for this demo:</p>
<pre class="bash"><code>#!/bin/bash

saw count \
    --id=workshop_demo \
    --sn=C04042E3 \
    --chip-mask=&quot;/path/to/C04042E3.barcodeToPos.h5&quot; \
    --omics=transcriptomics \
    --kit-version=&quot;Stereo-seq T FF V1.3&quot; \
    --sequencing-type=&quot;PE75_50+100&quot; \
    --organism=mouse \
    --tissue=brain \
    --fastqs=/path/to/fastq/directory/ \
    --image-tar=&quot;/path/to/C04042E3_SC_20240925_142342_4.1.0.tar.gz&quot; \
    --reference=/path/to/reference/output</code></pre>
<p>Notes:</p>
<ul>
<li><code>--reference</code> here should point to the directory
specified in the <code>--genome</code> parameter of
<code>saw makeRef</code></li>
<li><code>--image-tar</code> accepts image files processed by <a
href="#stereomap">StereoMap</a> or ImageStudio, and is optional</li>
<li><code>--fastqs</code> points to a directory containing fastq file
pairs. More than one pair can be present (e.g. if multiple sequencing
runs or lanes were used)</li>
<li><code>--organism</code> and <code>--tissue</code> can be any
string</li>
</ul>
</div>
</div>
<div id="saw-outputs" class="section level1">
<h1>SAW outputs</h1>
<p>Once this is complete, an <code>outs</code> directory will be created
similar to this:</p>
<pre class="bash"><code>$ tree outs/
.
├── analysis
│   ├── C04042E3.bin20_1.0.h5ad
│   ├── C04042E3.bin20_1.0.marker_features.csv
│   ├── C04042E3.bin50_1.0.h5ad
│   ├── C04042E3.bin50_1.0.marker_features.csv
│   ├── C04042E3.cellbin_1.0.adjusted.h5ad
│   ├── C04042E3.cellbin_1.0.adjusted.marker_features.csv
│   ├── C04042E3.cellbin_1.0.h5ad
│   └── C04042E3.cellbin_1.0.marker_features.csv
├── bam
│   └── annotated_bam
│       ├── C04042E3.Aligned.sortedByCoord.out.merge.q10.dedup.target.bam
│       └── C04042E3.Aligned.sortedByCoord.out.merge.q10.dedup.target.bam.csi
├── C04042E3.report.tar.gz
├── feature_expression
│   ├── C04042E3.adjusted.cellbin.gef
│   ├── C04042E3.cellbin.gef
│   ├── C04042E3.merge.barcodeReadsCount.txt
│   ├── C04042E3_raw_barcode_gene_exp.txt
│   ├── C04042E3.raw.gef
│   └── C04042E3.tissue.gef
├── image
│   ├── C04042E3_HE_mask_edm_dis_10.tif
│   ├── C04042E3_HE_mask.tif
│   ├── C04042E3_HE_regist.tif
│   └── C04042E3_HE_tissue_cut.tif
├── visualization.tar.gz
└── visualization.tar.gz.md5</code></pre>
<p>A detailed guide to these outputs can be found <a
href="https://stereotoolss-organization.gitbook.io/saw-user-manual-v8.1/analysis/outputs/count-outputs">in
the SAW manual</a>.</p>
<div id="html-report" class="section level2">
<h2>HTML report</h2>
<p>The first place to look is generally the HTML report. We can extract
it from the archive and open it in a browser:</p>
<pre class="r"><code># First extract the archive if needed

if (!file.exists(&#39;data/report/&#39;)) {
  untar(&#39;data/C04042E3.report.tar.gz&#39;, exdir=&#39;data&#39;)
}

# Now access it from data/report/report.html</code></pre>
<p>A more detailed guide to the SAW report can be found <a
href="https://stereotoolss-organization.gitbook.io/saw-user-manual-v8.1/analysis/outputs/html-report">in
the SAW manual</a>.</p>
</div>
<div id="gef-files" class="section level2">
<h2>GEF files</h2>
<p>These are the main count matrices, in HDF5 format.</p>
</div>
<div id="h5ad-files" class="section level2">
<h2>h5ad files</h2>
<p>These contain clustering results, for the most part. The default
method is Leiden clustering.</p>
</div>
<div id="visualization.tar.gz" class="section level2">
<h2>visualization.tar.gz</h2>
<p>This packages the main outputs into a single file for visualization.
It is also a handy file for distributing results easily, as it packages
most of the main outputs (excluding the BAM files).</p>
<pre class="r"><code># Extracting the visualization archive

if (!file.exists(&#39;data/visualization/&#39;)) {
  untar(&#39;data/visualization.tar.gz&#39;, exdir=&#39;data&#39;)
}</code></pre>
<p>Taking a look inside, we see many of the other output files have been
repackaged here:</p>
<pre class="bash"><code>.
├── C04042E3.adjusted.cellbin.gef
├── C04042E3.bin20_1.0.h5ad
├── C04042E3.bin50_1.0.h5ad
├── C04042E3.cellbin_1.0.adjusted.h5ad
├── C04042E3.rpi
├── C04042E3_SC_20240925_142342_4.1.0.tar.gz
├── C04042E3.stereo
├── C04042E3.tissue.gef
└── HE_matrix_template.txt</code></pre>
<p>A JSON manifest of the experiment can be found in
<code>.stereo</code>, which allows loading all of the packaged data in a
single step from StereoMap.</p>
<p>Although these have varied extensions, they’re mostly just hdf5
format archives.</p>
<p>The rpi file can be converted to tif files using SAW (although it’s
easier to just take the output directly from <code>saw count</code>)</p>
</div>
</div>
<div id="stereomap" class="section level1">
<h1>StereoMap</h1>
<p>StereoMap is an interactive tool to examine your Stereo-seq data
(similar to Loupe Browser from 10x).</p>
<p>StereoMap can be downloaded from STOmics directly: <a
href="https://en.stomics.tech/products/stomics-software/stomics-offline-software/list.html"
class="uri">https://en.stomics.tech/products/stomics-software/stomics-offline-software/list.html</a></p>
<p>StereoMap v4.1 is compatible with SAW v8.1. Previous versions also
included image QC in a seperate program, ImageStudio, which is now
integrated with StereoMap.</p>
</div>
<div id="stereopy" class="section level1">
<h1>StereoPy</h1>
<p>StereoPy is a Python library designed to work with Stereo-seq
data.</p>
<p><a
href="https://stereopy.readthedocs.io/en/latest/content/01_Usage_principles.html"
class="uri">https://stereopy.readthedocs.io/en/latest/content/01_Usage_principles.html</a></p>
<p>We will use it to investigate our data over on JupyterLab.</p>
</div>
<div id="loading-in-spatialexperiment" class="section level1">
<h1>Loading in SpatialExperiment</h1>
<p><a
href="https://www.bioconductor.org/packages/release/bioc/vignettes/SpatialExperiment/inst/doc/SpatialExperiment.html">SpatialExperiment</a>
library is an S4 class that extends SingleCellExperiment. While there’s
no way to import directly, we can find all the data we need from the
h5ad files provided from SAW.</p>
<p>First let’s load the matrix.</p>
<pre class="r"><code>library(rhdf5)
library(Matrix)

# Load the matrix data (CSR format)
matrix_data &lt;- h5read(&quot;data/visualization/C04042E3.bin20_1.0.h5ad&quot;, &quot;X&quot;)

# Extract dimensions (this is stored as an attribute)
shape &lt;- h5readAttributes(&quot;data/visualization/C04042E3.bin20_1.0.h5ad&quot;, &quot;X&quot;)$shape

# Extract the components
data &lt;- matrix_data$data        # Non-zero values
indices &lt;- matrix_data$indices  # Column indices of non-zero values
indptr &lt;- matrix_data$indptr    # Row pointers for non-zero values

# Reconstruct the CSR matrix (have to add 1 to indices for R)
csr_matrix &lt;- sparseMatrix(
  i = as.integer(rep(seq_len(length(indptr)-1), diff(indptr))),  # expand row indices
  j = as.integer(indices + 1),  # R uses 1-based indexing, so add 1 to column indices
  x = as.numeric(data),
  dims = shape,
)</code></pre>
<p>We can also add in row and col data.</p>
<pre class="r"><code>obs &lt;- h5read(&quot;data/visualization/C04042E3.bin20_1.0.h5ad&quot;, &quot;obs&quot;)
var &lt;- h5read(&quot;data/visualization/C04042E3.bin20_1.0.h5ad&quot;, &quot;var&quot;)

library(S4Vectors)</code></pre>
<pre><code>Loading required package: stats4</code></pre>
<pre><code>Loading required package: BiocGenerics</code></pre>
<pre><code>
Attaching package: &#39;BiocGenerics&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    IQR, mad, sd, var, xtabs</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    anyDuplicated, aperm, append, as.data.frame, basename, cbind,
    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,
    get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,
    match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,
    Position, rank, rbind, Reduce, rownames, sapply, saveRDS, setdiff,
    table, tapply, union, unique, unsplit, which.max, which.min</code></pre>
<pre><code>
Attaching package: &#39;S4Vectors&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:Matrix&#39;:

    expand, unname</code></pre>
<pre><code>The following object is masked from &#39;package:utils&#39;:

    findMatches</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    expand.grid, I, unname</code></pre>
<pre class="r"><code>library(SpatialExperiment)</code></pre>
<pre><code>Loading required package: SingleCellExperiment</code></pre>
<pre><code>Loading required package: SummarizedExperiment</code></pre>
<pre><code>Loading required package: MatrixGenerics</code></pre>
<pre><code>Loading required package: matrixStats</code></pre>
<pre><code>
Attaching package: &#39;MatrixGenerics&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:matrixStats&#39;:

    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,
    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,
    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,
    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,
    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,
    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,
    colWeightedMeans, colWeightedMedians, colWeightedSds,
    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,
    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,
    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,
    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,
    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,
    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,
    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,
    rowWeightedSds, rowWeightedVars</code></pre>
<pre><code>Loading required package: GenomicRanges</code></pre>
<pre><code>Loading required package: IRanges</code></pre>
<pre><code>Loading required package: GenomeInfoDb</code></pre>
<pre><code>Loading required package: Biobase</code></pre>
<pre><code>Welcome to Bioconductor

    Vignettes contain introductory material; view with
    &#39;browseVignettes()&#39;. To cite Bioconductor, see
    &#39;citation(&quot;Biobase&quot;)&#39;, and for packages &#39;citation(&quot;pkgname&quot;)&#39;.</code></pre>
<pre><code>
Attaching package: &#39;Biobase&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:MatrixGenerics&#39;:

    rowMedians</code></pre>
<pre><code>The following objects are masked from &#39;package:matrixStats&#39;:

    anyMissing, rowMedians</code></pre>
<pre class="r"><code># create SpatialExperiment object
spe &lt;- SpatialExperiment(
  assays = list(counts = t(csr_matrix)), 
  colData = DataFrame(cluster=as.factor(obs$leiden$codes), in_tissue=TRUE), # get cluster in there 
  rowData = DataFrame(gene_id=var$`_index`),
  spatialCoords = data.matrix(DataFrame(x=obs$x, y=obs$y))
)</code></pre>
<p>We can now use this as we would any other spot-based experiment.</p>
<pre class="r"><code>library(ggspavis)</code></pre>
<pre><code>Loading required package: ggplot2</code></pre>
<pre><code>Registered S3 method overwritten by &#39;ggside&#39;:
  method from   
  +.gg   ggplot2</code></pre>
<pre class="r"><code># Get norm counts 
colData(spe)$sums &lt;- colSums(counts(spe))
plotSpots(spe, annotate=&quot;sums&quot;)</code></pre>
<p><img src="figure/SAW.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
