<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Part 1</title>

<script src="site_libs/header-attrs-2.28/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cerulean.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
}

</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Stereo-seq workshop</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="setup.html">Setup</a>
</li>
<li>
  <a href="download.html">Download</a>
</li>
<li>
  <a href="SAW.html">Part 1</a>
</li>
<li>
  <a href="downstream.html">Part 2</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/sagc-bioinformatics/abacbs2024_workshop">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Part 1</h1>

</div>


<div id="stereo-seq-background" class="section level1">
<h1>Stereo-seq background</h1>
<p>In Stereo-seq builds upon the DNBSEQ chips that are already used in
MGI DNA nanoball sequencing (DNBSEQ). For background on DNBSEQ, see <a
href="https://mgi-tech.eu/technology">here</a>.</p>
<p>The chips are patterned, with fixed spots of 220 nm and a distance of
500 or 715 nm.</p>
<p>The first step is to deposit DNB templates with random barcodes on
the chip, which is sequenced to obtain the CID (coordinate ID) of each
DNB. This is the “mask file”, which is basically just a big matrix which
barcodes correspond to which locations, and is required to run SAW.</p>
<p>Then, oligos with MIDs (molecular IDs) and polyT sequences are
ligated to each spot.</p>
<p>Next, the tissue is loaded onto the chip, followed by fixation and
permeabilization to capture polyA RNAs.</p>
<p>Finally, cDNA is generated via reverse transcription and
amplification and used as a template for library prep.</p>
<p>More information on Stereo-seq can be found in the <a
href="%22https://www.cell.com/cell/fulltext/S0092-8674(22)00399-3%22">paper</a></p>
<div id="available-versions" class="section level2">
<h2>Available versions</h2>
<ul>
<li>Current fresh-frozen kit version is v1.3</li>
<li>OMNI is the new FFPE offering, which also uses total RNA</li>
</ul>
</div>
</div>
<div id="processing-data-with-saw" class="section level1">
<h1>Processing data with SAW</h1>
<p>Once your experiment is complete, you will have the following:</p>
<ul>
<li>Mask file</li>
<li>Fastq files</li>
</ul>
<p>You will need to process the data using Stereo-seq Analysis Workflow
(SAW).</p>
<p>The latest version of SAW is 8.1, with reference documentation found
here:</p>
<p><a
href="https://stereotoolss-organization.gitbook.io/saw-user-manual-v8.1"
class="uri">https://stereotoolss-organization.gitbook.io/saw-user-manual-v8.1</a></p>
<ul>
<li>We’re going to see where to download SAW and how to run it</li>
</ul>
<div id="obtaining-saw" class="section level2">
<h2>Obtaining SAW</h2>
<p>Next, I’m going to show you how to use SAW to:</p>
<ul>
<li>check your annotation is valid</li>
<li>build a reference (I’m using GRCm39 from GENCODE)</li>
</ul>
</div>
<div id="saw-makeref" class="section level2">
<h2>SAW makeRef</h2>
<p>SAW v8.1 is quite easy to use, but not yet very configurable. To
build the reference for this example, here’s the command I used:</p>
<pre class="bash"><code>#!/bin/bash

saw makeRef \
    --mode=STAR \
    --fasta=/path/to/GRCm39.genome.fa \
    --gtf=/path/to/annotation.gtf \
    --genome=/path/to/reference/output</code></pre>
<p>Notes:</p>
<ul>
<li>The GENCODE GRCm39 reference was used</li>
<li>The <code>--genome</code> parameter actually specifies the name of
the output directory</li>
<li>There is no way to pass parameters to STAR in this version.</li>
</ul>
<p>To have more control over the parameters used, you can use version
7.x of SAW, which is <a href="https://github.com/STOmics/SAW">available
on github</a>.</p>
</div>
<div id="saw-count" class="section level2">
<h2>SAW count</h2>
<pre class="bash"><code>#!/bin/bash

saw count \
    --id=workshop_demo \
    --sn=C04042E3 \
    --chip-mask=&quot;/path/to/C04042E3.barcodeToPos.h5&quot; \
    --omics=transcriptomics \
    --kit-version=&quot;Stereo-seq T FF V1.3&quot; \
    --sequencing-type=&quot;PE75_50+100&quot; \
    --organism=mouse \
    --tissue=brain \
    --fastqs=/path/to/fastq/directory/ \
    --image-tar=&quot;/path/to/C04042E3_SC_20240925_142342_4.1.0.tar.gz&quot; \
    --reference=/path/to/reference/output</code></pre>
<p>Notes:</p>
<ul>
<li><code>--reference</code> here should point to the directory
specified in the <code>--genome</code> parameter of
<code>saw makeRef</code></li>
<li><code>--image-tar</code> accepts image files processed by <a
href="#stereomap">StereoMap</a> or ImageStudio, and is optional</li>
<li><code>--fastqs</code> points to a directory containing fastq file
pairs. More than one pair can be present (e.g. if multiple sequencing
runs or lanes were used)</li>
<li><code>--organism</code> and <code>--tissue</code> can be any
string</li>
</ul>
</div>
<div id="next-steps" class="section level2">
<h2>Next steps</h2>
<ul>
<li>Description of what you get back (compare SAW versions)
<ul>
<li>mention where you can get it, why not here</li>
</ul></li>
<li>Take us through the example dataset</li>
</ul>
<p><a
href="https://stereotoolss-organization.gitbook.io/saw-user-manual-v8.1/tutorials/run-main-pipeline/stereo-seq-ff"
class="uri">https://stereotoolss-organization.gitbook.io/saw-user-manual-v8.1/tutorials/run-main-pipeline/stereo-seq-ff</a></p>
</div>
</div>
<div id="stereomap" class="section level1">
<h1>StereoMap</h1>
<p><a
href="https://en.stomics.tech/products/stomics-software/stomics-offline-software/list.html"
class="uri">https://en.stomics.tech/products/stomics-software/stomics-offline-software/list.html</a></p>
</div>
<div id="stereopy" class="section level1">
<h1>StereoPy</h1>
<p><a
href="https://stereopy.readthedocs.io/en/latest/content/01_Usage_principles.html"
class="uri">https://stereopy.readthedocs.io/en/latest/content/01_Usage_principles.html</a></p>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
