---
title: "Part 1"
site: workflowr::wflow_site
editor_options:
  chunk_output_type: console
---

# Stereo-seq background

In Stereo-seq builds upon the DNBSEQ chips that are already used in MGI DNA nanoball sequencing (DNBSEQ). For background on DNBSEQ, see [here](https://mgi-tech.eu/technology).

The chips are patterned, with fixed spots of 220 nm and a distance of 500 or 715 nm.

The first step is to deposit DNB templates with random barcodes on the chip, which is sequenced to obtain the CID (coordinate ID) of each DNB. This is the “mask file”, which is basically just a big matrix which barcodes correspond to which locations, and is required to run SAW. 
 

Then, oligos with MIDs (molecular IDs) and polyT sequences are ligated to each spot.  

Next, the tissue is loaded onto the chip, followed by fixation and permeabilization to capture polyA RNAs.

Finally, cDNA is generated via reverse transcription and amplification and used
as a template for library prep.

More information on Stereo-seq can be found in the [paper]("https://www.cell.com/cell/fulltext/S0092-8674(22)00399-3")


## Available versions

- Current fresh-frozen kit version is v1.3
- OMNI is the new FFPE offering, which also uses total RNA

# Processing data with SAW

Once your experiment is complete, you will have the following:

- Mask file
- Fastq files

You will need to process the data using Stereo-seq Analysis Workflow (SAW).

The latest version of SAW is 8.1, with reference documentation found here:

https://stereotoolss-organization.gitbook.io/saw-user-manual-v8.1

- We're going to see where to download SAW and how to run it

## Obtaining SAW



Next, I'm going to show you how to use SAW to:

- check your annotation is valid
- build a reference (I'm using GRCm39 from GENCODE)


## SAW makeRef

SAW v8.1 is quite easy to use, but not yet very configurable.
To build the reference for this example, here's the command I used:

```{bash, eval=FALSE}
#!/bin/bash

saw makeRef \
    --mode=STAR \
    --fasta=/path/to/GRCm39.genome.fa \
    --gtf=/path/to/annotation.gtf \
    --genome=/path/to/reference/output
```

Notes:

- The GENCODE GRCm39 reference was used
- The `--genome` parameter actually specifies the name of the output directory
- There is no way to pass parameters to STAR in this version.

To have more control over the parameters used, you can use version 7.x of SAW,
which is [available on github](https://github.com/STOmics/SAW).


## SAW count

```{bash, eval=FALSE}
#!/bin/bash

saw count \
    --id=workshop_demo \
    --sn=C04042E3 \
    --chip-mask="/path/to/C04042E3.barcodeToPos.h5" \
    --omics=transcriptomics \
    --kit-version="Stereo-seq T FF V1.3" \
    --sequencing-type="PE75_50+100" \
    --organism=mouse \
    --tissue=brain \
    --fastqs=/path/to/fastq/directory/ \
    --image-tar="/path/to/C04042E3_SC_20240925_142342_4.1.0.tar.gz" \
    --reference=/path/to/reference/output
```

Notes:

- `--reference` here should point to the directory specified in the `--genome` parameter of `saw makeRef`
- `--image-tar` accepts image files processed by [StereoMap](#stereomap) or ImageStudio, and is optional
- `--fastqs` points to a directory containing fastq file pairs. More than one pair can be present (e.g. if multiple sequencing runs or lanes were used) 
- `--organism` and `--tissue` can be any string

## Next steps

- Description of what you get back (compare SAW versions)
  - mention where you can get it, why not here
- Take us through the example dataset

https://stereotoolss-organization.gitbook.io/saw-user-manual-v8.1/tutorials/run-main-pipeline/stereo-seq-ff

# StereoMap

# StereoPy
